src/components/tx/ApprovalEditor/ApprovalEditorForm.tsx

```diff
@@ -29,44 +30,36 @@ export const ApprovalEditorForm = ({
     mode: 'onChange',
   })
 
-  const {
-    formState: { errors, dirtyFields },
-    getValues,
-    reset,
-  } = formMethods
+  const { getValues, reset } = formMethods
 
   const onSave = () => {
     const formData = getValues('approvals')
     updateApprovals(formData)
     reset({ approvals: formData })
   }
 
+  let fieldIndex = 0
+
   return (
     <FormProvider {...formMethods}>
       <List className={css.approvalsList}>
-        {approvalInfos.map((tx, idx) => (
-          <ListItem
-            key={tx.tokenAddress + tx.spender}
-            className={0n === tx.amount ? css.zeroValueApproval : undefined}
-            disablePadding
-            data-testid="approval-item"
-          >
-            <ApprovalItem spender={tx.spender} method={tx.method}>
-              <>
-                <ApprovalValueField name={`approvals.${idx}`} tx={tx} />
-                <Track {...MODALS_EVENTS.EDIT_APPROVALS}>
-                  <IconButton
-                    className={css.iconButton}
-                    onClick={onSave}
-                    disabled={!!errors.approvals || !dirtyFields.approvals?.[idx]}
-                    title="Save"
-                  >
-                    <SvgIcon component={CheckIcon} />
-                  </IconButton>
-                </Track>
-              </>
-            </ApprovalItem>
-          </ListItem>
+        {Object.entries(groupedApprovals).map(([spender, approvals], spenderIdx) => (
+          <Box key={spender}>
+            <Stack gap={2}>
+              <SpenderField address={spender} />
+              {approvals.map((tx) => (
+                <ListItem
+                  key={tx.tokenAddress + tx.spender}
+                  className={0n === tx.amount ? css.zeroValueApproval : undefined}
+                  disablePadding
+                  data-testid="approval-item"
+                >
+                  <EditableApprovalItem approval={tx} name={`approvals.${fieldIndex++}`} onSave={onSave} />
+                </ListItem>
+              ))}
+            </Stack>
+            {spenderIdx !== Object.keys(groupedApprovals).length - 1 && <Divider />}
```

Comment:
This Divider needs to be moved into the `Stack` component above so that it has spacing
<img width="635" alt="Screenshot 2024-04-26 at 14 28 35" src="https://github.com/safe-global/safe-wallet-web/assets/5880855/6f2756ae-c45b-47be-a924-ac2f5c3ab160">